### PR TITLE
[RTI-8301+RTI-8302] Creating geodata2:is_ipv6_mmdb/0 to support new magellan's IPv4+IPv6 MMDB file

### PR DESCRIPTION
**Pull request by diegomanuel**
## Tickets: [RTI-8301](https://adroll.atlassian.net/browse/RTI-8301) + [RTI-8302](https://adroll.atlassian.net/browse/RTI-8302)

Used at `boodah_geo:lookup/1` to support new `magellan`'s **IPv4+IPv6** `MMDB` file.
Adding `geodata2:is_ipv6_mmdb/0` and `geodata2_format:is_ipv6/1` in order to be able to know from the "outside world" if the loaded DB is `v4` or `v6`.

For that, I kept the version in app env (using `application:set_env/2` and `application:get_env/3`) instead of storing it inside `geodata2` process state.
This is a **temporary** helper fuction in order to be **backward compatible** with `IPv4` DB, that could/should be removed once the `IPv6` DB is definitely used.

## Related PRs:
- https://github.com/SemanticSugar/magellan/pull/158
- https://github.com/SemanticSugar/erl_utils/pull/47
- https://github.com/SemanticSugar/boodah/pull/4491

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-REBAR ?= $(shell which rebar 2>/dev/null || which ./rebar)
+REBAR ?= $(shell which rebar3 2>/dev/null || which rebar 2>/dev/null || which ./rebar)
```
Comment (elbrujohalcon) [inline Makefile:1]: Can we just remove support for the old `rebar` entirely, please?

File: README.md [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
-This is a forked branch of [brigadier/geodata2](https://github.com/brigadier/geodata2), which uses a shared ets table to hold the database state rather than a cluster of workers. This was done to improve performance.
+This is a forked branch of [altenmueller/geodata2](https://github.com/altenmueller/geodata2), which uses a shared `ETS` table to hold the database state rather than a cluster of workers.  
+This was done to improve performance.
 
 #### Features:
-* Supports all types (hopefully) of bases in .mmdb format, both IPv4 and IPv6 ones;
+* Supports all types (hopefully) of bases in .mmdb format, both IPv4 and IPv6 ones.
 * Safe from binary reference leakage, binary parts are getting copied.
-* R18 compatible
+* R18 compatible.
```
Comment (elbrujohalcon) [inline README.md:9]: It should be OTP22 compatible, at least.

File: src/geodata2.erl [hunk 3]
```diff
@@ -91,5 +93,13 @@ get_env(App, Key) ->
 id(X) ->
     X.
 
+%% @TODO [RTI-8302] This one could be removed after the IPv4+IPv6 MMDB is definitely used
+is_ipv6_mmdb() ->
+    application:get_env(geodata2, ipv6, false).
+
+%% @TODO [RTI-8302] This one could be removed after the IPv4+IPv6 MMDB is definitely used
```
Comment (elbrujohalcon) [inline src/geodata2.erl:100]: This library is public (it's a fork from somebody else) and those jira tickets are private. I'm not sure if we should leave them here.
As a matter of fact, we might eventually send a PR upstream… and we'll definitely not want them there then.

**Review by elbrujohalcon — CHANGES_REQUESTED** [review]
PR is fine, but since we need to change things to support `rebar3`, I would vote for just remove support for the old `rebar` instead of keeping both.
And since this is a forked public library… I would be wary of disclosing our JIRA ticket numbers.

File: Makefile (hunk 1 above)
Comment (elbrujohalcon) [reply Makefile:1]: Or just not change the `Makefile` and `README.md` files at all since this is a forked public library.

Comment (diegomanuel) [reply Makefile:1]: Sure, will remove the old `rebar`.

Comment (diegomanuel) [reply Makefile]: Done!

File: README.md (hunk 2 above)
Comment (diegomanuel) [reply README.md:9]: Is that ok? I just added:
* OTP22 compatible.
* Uses rebar3.

**Review by elbrujohalcon — APPROVED** [review]

**Pull request merged**